pkg/deployment/deployment_item.go

```diff
@@ -47,6 +48,55 @@ type DeploymentItem struct {
 	renderedYamlPath      string
 }
 
+// TODO: find better name then DeploymentItem...Vars?
+// IDEA: Make these sub-objects of actual di/dp
+type DeploymentItemProjectVars struct { // info about the deployment.yaml project that included this item
+	Parent  *DeploymentItemProjectVars // <parent project info, same structure as `project`>
+	DirName string                     `yaml:"dirName" validate:"required"` // dirname
+	RelPath []string                   // <relative directory to current source>
```

Comment:
Why an array of path elements instead the full path as a single string?

Reply:
that is sure wort of short discussion, 

1. I assumed 

```
cluster/appsStage0/argo/kustomization.yml
1:namespace: &ns {{ this.revPath[0] }}
````

this is "more accessible/readable" and technically same code for `go/jinja` templates then:

```
cluster/appsStage0/argo/kustomization.yml
1:namespace: &ns {{ this.revPath.split('/')[0] }}
```

or using `.index() or |reverse`

1. why `relPath`, `revPath` -> assume this

`targets/target_name/deployment_stage/namespace/app_instance`

  * ^^ there are times where you more need (app_instance)
  * ^^ there are times where you more want some "beginning" like `target_name` or in sub-deployments it could be `deployment_stage`
 
but in both cases you know it;s first/last but you dont want to "count" where in the midle the value you want is (nor do the parsing in templates, multiple times).

Overall, I would go with what you will decide, string/array are still pretty much OK for me for PR.

Reply:
Why not simply `this.relPath[-2]` to get the second item from the end?

Reply:
I clearly prefer the plain string version and let people then use Jinja2 templating to split it if needed. I understand it's more repetitive, but its very explicit and doesn't hide any magic from whoever is reading it.

Reply:
ok, will respect that and change to string, np